### ISSUE TITLE
Unnecessary line splitting in pattern

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#1107]
```
-                        (None, _) | (_, Some(hir::MutMutable)) => {
+                        (None, _) |
+                        (_, Some(hir::MutMutable)) => {
```

**Comment by sinkuu** [comment]
This seems because patterns are [formatted with `DefinitiveListTactic::Vertical`](https://github.com/rust-lang-nursery/rustfmt/blob/664bf3eae707954b4bc5cb818d8a7d780e7c3c95/src/expr.rs#L1107) if [pat_is_simple](https://github.com/rust-lang-nursery/rustfmt/blob/664bf3eae707954b4bc5cb818d8a7d780e7c3c95/src/expr.rs#L1217) returns false.

**Comment by topecongiro** [comment]
Closed via #1775.
